src/main/java/org/mockbukkit/mockbukkit/entity/SpawnerMinecartMock.java

```diff
@@ -45,145 +60,135 @@ public SpawnerMinecartMock(@NotNull ServerMock server, @NotNull UUID uuid)
 	@Override
 	public @Nullable EntityType getSpawnedType()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return this.spawnerType;
 	}
 
 	@Override
 	public void setSpawnedType(@Nullable EntityType entityType)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkArgument(entityType != EntityType.UNKNOWN, "Can't spawn EntityType %s from mob spawners!", entityType);
+		this.spawnerType = entityType;
 	}
 
 	@Override
 	public int getDelay()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return this.spawnDelay;
 	}
 
 	@Override
-	public int getRequiredPlayerRange()
+	public void setDelay(int delay)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		this.spawnDelay = delay;
 	}
 
 	@Override
-	public void setRequiredPlayerRange(int i)
+	public int getMinSpawnDelay()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return this.minSpawnDelay;
 	}
 
 	@Override
-	public int getSpawnRange()
+	public void setMinSpawnDelay(int spawnDelay)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkArgument(spawnDelay <= this.getMaxSpawnDelay(), "Minimum Spawn Delay must be less than or equal to Maximum Spawn Delay");
+		this.minSpawnDelay = spawnDelay;
```

Comment:
Negative values?

Reply:
In paper and Minecraft i did not find any validation for that